src/repositories/friendship.repository.ts

```diff
@@ -17,6 +17,122 @@ export class FriendshipRepository extends Repository {
     return this.change('destroy', id, mediaIdAttribution);
   }
 
+  async removeFriendship(id: string | number, mediaIdAttribution?: string) {
+    return this.change('remove_follower', id, mediaIdAttribution);
+  }
+
+  async muteUserMedia(id: string | number, option: string, mediaIdAttribution?: string) {
+    var additional = {};
+    switch (option) {
+      case 'story':
+        additional['target_reel_author_id'] = id;
+        break;
+      case 'post':
+        additional['target_posts_author_id'] = id;
+        break;
+      default:
+        throw new Error('Invalid option: choose story or post');
+    }
+    var defaultParams = {
+      _csrftoken: this.client.state.cookieCsrfToken,
+      user_id: id,
+      radio_type: this.client.state.radioType,
+      _uid: this.client.state.cookieUserId,
+      device_id: this.client.state.deviceId,
+      _uuid: this.client.state.uuid,
+      media_id_attribution: mediaIdAttribution,
+    };
+    const { body } = await this.client.request.send({
+      url: `/api/v1/friendships/mute_posts_or_story_from_follow/`,
+      method: 'POST',
+      form: this.client.request.sign({ ...defaultParams, ...additional }),
+    });
+    return body.friendship_status;
+  }
+
+  async unmuteUserMedia(id: string | number, option: string, mediaIdAttribution?: string) {
+    var additional = {};
+    switch (option) {
+      case 'story':
+        additional['target_reel_author_id'] = id;
+        break;
+      case 'post':
+        additional['target_posts_author_id'] = id;
+        break;
+      default:
+        throw new Error('Invalid option: choose story or post');
+    }
+    var defaultParams = {
+      _csrftoken: this.client.state.cookieCsrfToken,
+      user_id: id,
+      radio_type: this.client.state.radioType,
+      _uid: this.client.state.cookieUserId,
+      device_id: this.client.state.deviceId,
+      _uuid: this.client.state.uuid,
+      media_id_attribution: mediaIdAttribution,
+    };
+    const { body } = await this.client.request.send({
+      url: `/api/v1/friendships/unmute_posts_or_story_from_follow/`,
+      method: 'POST',
+      form: this.client.request.sign({ ...defaultParams, ...additional }),
+    });
+    return body.friendship_status;
+  }
+
+  async favorite(id: string | number, mediaIdAttribution?: string) {
+    return this.change('favorite', id, mediaIdAttribution);
+  }
+
+  async unfavorite(id: string | number, mediaIdAttribution?: string) {
+    return this.change('unfavorite', id, mediaIdAttribution);
+  }
+
+  async block(id: string | number, mediaIdAttribution?: string) {
+    return this.change('block', id, mediaIdAttribution);
+  }
+
+  async blockFromStory(id: string | number, mediaIdAttribution?: string) {
```

Comment:
`blockFriendReel`
If you want, you could describe it in comments using human language using
https://github.com/microsoft/tsdoc